### PR TITLE
Change pride tiles to generic holiday tiles, add Tramstation support

### PR DESCRIPTION
**Pull request by lessthnthree**
## About The Pull Request

- Rewrites the Pride Week rainbow tiles coloring into a more generic 'Holiday' tile coloring proc so that it can be used by any holiday
- Adds support for different color patterns
- Adds support for the trimline decal subtype Tramstation uses
- Adds stack types for tram tiles so that if you pull them up you can put them back
- Fixes naming of tile decals

## Why It's Good For The Game

- The generic proc allows us to specify a different color set for each holiday (think red white blue for July 4, or red/white/green for Christmas)
- Players now have a way to put back tram platform tiles that have been pulled up
- Tramstation gets colored tiles like the other maps

image

## Changelog

:cl: LT3
fix: Tram platform tiles now have proper stack types
refactor: Pride week tile coloring refactored into a generic holiday decal system
spellcheck: Tile decals are no longer all named 'corner'
/:cl:

### CODE REVIEW

**Comment by TheVekter** [comment]
Are pride tiles still the same colors they used to be?

**Comment by lessthnthree** [comment]
> Are pride tiles still the same colors they used to be?

Yes, there's no change to the defined colors.

File: code/game/objects/effects/decals/misc.dm [hunk 1]
```diff
@@ -89,3 +89,14 @@
 	icon = 'icons/obj/smooth_structures/lattice.dmi'
 	icon_state = "lattice-255"
 	density = TRUE
+
+/// Given an atom, will return what color it should be to match the event/holiday
+/obj/effect/proc/get_holiday_color(atom/thing_to_color, axis = STRIPE_HORIZONTAL)
```
Comment (MrMelbert) [inline code/game/objects/effects/decals/misc.dm:94]: I'm not sure you want this to be a proc on `/obj/effect`, given it's copy+pasted onto `/turf`

Maybe you should instead bump it down a level from `/datum/holiday/pride_week/proc/get_floor_tile_color` to `/datum/holiday/proc/get_floor_tile_color`, which then each holiday can override on a subtype basis to do their own custom coloring if desired

File: code/game/objects/effects/decals/turfdecal/tilecoloring.dm [hunk 2]
```diff
@@ -4,22 +4,22 @@
...
+	if(check_holidays(APRIL_FOOLS))
 		color = "#[random_short_color()]"
-	else if (check_holidays(PRIDE_WEEK))
-		var/datum/holiday/pride_week/pride_week = GLOB.holidays[PRIDE_WEEK]
-		color = pride_week.get_floor_tile_color(src)
 
+	else if(check_holidays(PRIDE_WEEK))
+		if(istype(src, /obj/effect/turf_decal/tile/neutral/tram))
+			color = get_holiday_color(src, STRIPE_VERTICAL)
+		else color = get_holiday_color(src, STRIPE_HORIZONTAL)
```
Comment (MrMelbert) [inline code/game/objects/effects/decals/turfdecal/tilecoloring.dm:17]: `istype(src)` usually is icky behavior, this should probably be a var on the effect that determines whether to use vertical or horizontal stripes

File: code/__DEFINES/turfs.dm [hunk 3]
```diff
@@ -94,3 +94,8 @@
 #define TURF_PATHING_PASS_PROC 1
 /// Turf is never passable
 #define TURF_PATHING_PASS_NO 2
+
+/// Generate horizontal stripes for turf decals
+#define STRIPE_HORIZONTAL 0
+/// Generate vertical stripes for turf decals
+#define STRIPE_VERTICAL 1
```
Comment (MrMelbert) [inline code/__DEFINES/turfs.dm:101]: Should mention these are only used for holiday coloring

**Review by MrMelbert — CHANGES_REQUESTED** [review]
Do you intend on making another holiday use rainbow tiles? If so I'm not sure this should really be a global

File: code/game/objects/effects/decals/misc.dm (hunk 1 above)
Comment (lessthnthree) [reply code/game/objects/effects/decals/misc.dm:94]: The decals are /objects/effects/decals/turfdecal, not a turf. My thought behind the global color list is its also used by the rainbow tile objects (which are rainbow year round, not just the one week.)

For other holidays they can override the list with their holiday specific colors.

File: code/__DEFINES/turfs.dm (hunk 3 above)
Comment (lessthnthree) [reply code/__DEFINES/turfs.dm:101]: Sure.

**Review by Twaticus — APPROVED** [review]

File: code/game/objects/effects/decals/turfdecal/tilecoloring.dm (hunk 2 above)
Comment (lessthnthree) [reply code/game/objects/effects/decals/turfdecal/tilecoloring.dm]: Yep, thanks for your help.

**Comment by lessthnthree** [comment]
There [MrMelbert](https://github.com/MrMelbert), redone to what we discussed yesterday.

**Review by Wallemations — APPROVED** [review]

**Comment by TheVekter** [comment]
Hooray! Pride is saved!

If God hates the gays then why do we keep winning

**Comment by optimumtact** [comment]
you can't add commits after it's merged

**Comment by lessthnthree** [comment]
I'm not, the two commits below were early on but it thinks that they're from ~the future~

**Comment by optimumtact** [comment]
interesting, becuase they claim to be have been made on april 1st lol

**Comment by lessthnthree** [comment]
I still have another 3 weeks for corners

image

**Pull request merged**